lib/dnsimple/error.rb

```diff
@@ -9,8 +9,19 @@ class RequestError < Error
 
     def initialize(http_response)
       @http_response = http_response
-      super(http_response.code.to_s)
+      super(message_from(http_response))
     end
+
+    private
+
+    def message_from(http_response)
+      if http_response.headers["Content-Type"] == "application/json"
+        http_response.parsed_response["message"]
+      else
+        http_response.code
```

Comment:
Also note that this is an edge case that shouldn't happen. The main goal here is to prevent the client from crashing when the response is not a JSON one.